### ISSUE TITLE
"app.search" function not getting more than 300 results

### ISSUE THREAD

**Issue by Fastios98** [mahrtayyab/tweety#162]
Hey thanks for the library
I have a problem that I run into when trying to get a big number of tweets, I used the following code
`data = app.search("iPad Pro", pages=5000, wait_time=[2, 4])`
but despite putting the number of pages so high I ended up with 181 tweets
I also tried other wait times like 10, and ended up with 300 results.
I am new to python so I may be missing something but it seems like the function is not behaving as intended.
I tried the iter_search as well and saw it in action, the first couple of pages work fine but then it starts slowing down then I get a Rate Limit Exceeded error
![Iter_Search_Tweety](https://github.com/mahrtayyab/tweety/assets/101416033/a482d0ea-23ca-4862-92f0-ae968042672d)
this is the output of the Iterative search function, you can see how it slows down as it goes through the pages until it stops after 20 or so pages.

**Comment by mahrtayyab** [comment]
Search has 50 requests per 15 minutes limit , slow down your requests

**Comment by Fastios98** [comment]
I just tried increasing the wait_time up to 500 and 600 and tried both the search and iter_search methods.
I am getting the same problem; this time it is not really exceeding 200 search results.
Maybe I wasn't clear in my initial comment, but most of the time it executes the code without the Rate Limit Exceeded Error
It just works fine but I still end up with a little less than 200 results each time. and looking at the iterative search I can see it start to fall off after the first 5 or 6 pages, exactly like the output I showed in the original comment.
Am I doing something wrong here and that's not how you slow down the requests?
Has anybody tried the Search method recently?
